src/app/(auth)/onboarding/page.tsx

```diff
@@ -4,7 +4,11 @@ import OnboardingForm from "@/components/shared/Form/Onboarding/OnboardingForm";
 import { getSession } from "~/auth/auth";
 import { redirect } from "next/navigation";
 
-const Onboarding = async ({ searchParams }: any) => {
+const Onboarding = async ({
+  searchParams,
+}: {
+  [key: string]: string | string[] | undefined;
+}) => {
```

Comment:
Let's take a closer look at this and the feedback.

![image](https://github.com/dalbanhi/git-note/assets/4108484/a3db8b38-82c1-4895-bbe1-412da0f4581a)

Our types have to explicitly state what key in the object we're typing. If you want to use the dynamic 'any key' version of the typing using `[key: string]` then you still need to write `searchParams` here:
```js
  searchParams: { [key: string]: number };
```

Your other components also expect a number- so a string will cause a typeerror. You can use my original suggestion or this code:

```js
const Onboarding = async ({
  searchParams,
}: {
  searchParams: { [key: string]: number };
}) => {
```

Otherwise we get this build error:

![image](https://github.com/dalbanhi/git-note/assets/4108484/d6e44859-e714-41ca-8543-0219311dee47)

Reply:
Hi Brandon, thanks for catching that. Maybe during our meeting today, we can talk about how to view those build errors-- I don't think I got it.

Reply:
That's actually great timing! I did want to spend some time today setting up your deployment so we can make sure each branch builds correctly without having to test manually.

But give this a shot:

Instead of running `npm run dev` to start your development server, instead run
```
npm run build
```
To start the build process. This will kick up any typescript errors (one's that our development server will ignore) and other issues that a deployment will face.


We can talk more in-depth in the meeting about disabling the linting errors on the build and how to approach fixing typescript and build errors as well as how to deploy this repo to vercel :)